### PR TITLE
docs: Add docs for running headscale on sealos

### PR DESCRIPTION
**Pull request by yangchuansheng**
- [x] read the [CONTRIBUTING guidelines](README.md#contributing)
- [ ] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [x] updated documentation if needed
- [ ] updated CHANGELOG.md

### CODE REVIEW

File: docs/running-headscale-sealos.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Running headscale in Sealos
+
+## Goal
+
+This documentation has the goal of showing a user how-to run `headscale` in [Sealos](https://sealos.io).
+
+Sealos is a cloud operating system distribution based on the Kubernetes kernel. It adopts a cloud-native approach, discarding traditional cloud computing architecture, and shifting towards a new architecture with Kubernetes as the cloud kernel. This allows enterprises to use the cloud as effortlessly as they would use a personal computer.
```
Comment (ohdearaugustin) [inline docs/running-headscale-sealos.md:7]: A bit much advertisement for my taste (:

**Review by ohdearaugustin — CHANGES_REQUESTED** [review]
Generally well written documentation. Just added some minor suggestions.

**Comment by kradalby** [comment]
I'm sceptical to having to specific documentation for niche distributions, but if it is community maintained, I suppose it might be fine.

What do you think @ohdearaugustin ?

**Comment by ohdearaugustin** [comment]
I think the same. I already asked to changes some things, therefore with disclaimer I don't see any problem with it.

**Comment by ohdearaugustin** [comment]
@yangchuansheng would you please fix the suggestion otherwise I will close the PR.

**Comment by yangchuansheng** [comment]
> @yangchuansheng would you please fix the suggestion otherwise I will close the PR.

Ok, I'll fix the suggestion soon

**Comment by yangchuansheng** [comment]
> @yangchuansheng would you please fix the suggestion otherwise I will close the PR.

done! Please kindly review the modifications and let me know if there's anything else that needs to be addressed or if you have any further suggestions.

File: docs/running-headscale-sealos.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    **It might be outdated and it might miss necessary steps**.
+
+## Goal
+
+This documentation has the goal of showing a user how-to run `headscale` on Sealos.
+
+## Running headscale server
+
+1. Click the following prebuilt template:
```
Comment (ohdearaugustin) [inline docs/running-headscale-sealos.md:16]: Maybe add which version you are installing.

**Review by ohdearaugustin — CHANGES_REQUESTED** [review]
Small changes, mainly wording

**Comment by yangchuansheng** [comment]
> Small changes, mainly wording

Done.

**Comment by yangchuansheng** [comment]
Please let me know if there's anything else I should work on or if you have any further feedback, thanks.

**Review by ohdearaugustin — APPROVED** [review]

**Pull request merged**